elections/uk/management/commands/uk_create_elections_from_every_election.py

```diff
@@ -0,0 +1,193 @@
+
+        try:
+            organization_extra = OrganizationExtra.objects.get(
+                slug=org_slug)
+            organization = organization_extra.base
+        except OrganizationExtra.DoesNotExist:
+            organization, _ = Organization.objects.get_or_create(
+                name=org_name,
+                defaults={'classification': classification})
+            organization_extra = OrganizationExtra.objects.create(
+                base=organization,
+                slug=org_slug
+            )
+        if not election_model.organization:
+            election_model.organization = organization_extra.base
+            election_model.save()
+        return organization_extra
+
+    def add_area(self, area_info, election, organization_extra):
+        if area_info['organisation']['territory_code'] == "NIR":
+            partyset_name = "Northern Ireland"
+            country = "ni"
+        else:
+            partyset_name = "Great Britain"
+            country = "gb"
+
+        party_set, _ = PartySet.objects.get_or_create(
+            slug=country, defaults={'name': partyset_name}
+        )
+
+        if area_info['division']:
+            area_type_name = area_info['division']['division_type']
+        else:
+            area_type_name = area_info['organisation']['organisation_type']
+
+        area_type, _ = AreaType.objects.update_or_create(
+            name=area_type_name,
+            defaults={'source': 'elections.democracyclub.org.uk'}
+        )
+
+        if not election.area_types.filter(
+                name=area_type.name).exists():
+            election.area_types.add(area_type)
+
+        if area_info['division']:
+            area_name = area_info['division']['name']
+            post_role = area_info['division']['geography_curie']
+            post_extra_slug = ":".join([
+                area_info['division']['division_type'],
+                area_info['division']['geography_curie'].split(':')[-1]
+            ])
+            area_identifier = area_info['division']['geography_curie']
+            winner_count = area_info['division']['seats_total']
+        else:
+            # TODO
+            area_name = area_info['organisation']['official_name']
+            post_role = area_info['elected_role']
+            post_extra_slug = area_info['organisation']['slug']
+            area_identifier = ":".join([
+                area_info['organisation']['organisation_type'],
+                area_info['organisation']['slug'],
+            ])
+            winner_count = 1
+
+        area, _ = Area.objects.update_or_create(
+            identifier=area_identifier,
+            defaults={
+                'name': area_name,
+                # 'parent': parent_area,
+            }
+        )
+
+        AreaExtra.objects.get_or_create(
+            base=area,
+            defaults={'type': area_type}
+        )
+
+        post, _ = Post.objects.update_or_create(
```

Comment:
I think the creation or update of the post has to be done a bit differently, I'm afraid, since effectively the primary key for a Post + PostExtra pair is the `slug` field on `PostExtra`. The code here risks unpairing those (I think!). The pattern for doing this safely (which should probably be extracted to a manager method to make this easier, e.g. [something like what we did with `ImageExtraManager`](https://github.com/DemocracyClub/yournextrepresentative/blob/c668ede5f5c25f93228238700403d071f04d6f4f/candidates/models/popolo_extra.py#L745-L780)  does) is:

https://github.com/DemocracyClub/yournextrepresentative/blob/c668ede5f5c25f93228238700403d071f04d6f4f/elections/cr/management/commands/cr_create_basic_site.py#L115-L138